packages/playwright-core/src/server/isomorphic/utilityScriptSerializers.ts

```diff
@@ -163,6 +177,11 @@ export function source() {
     if (typeof value === 'bigint')
       return { bi: value.toString() };
 
+    if (isMap(value))
+      return { m: serialize(Object.fromEntries(value), handleSerializer, visitorInfo) };
```

Comment:
I believe  this turns non-string keys into strings. I think we should do manual serialization: array of values for set, array `[key1, value1, key2, value2, ...]` for maps.

Reply:
resolved in: https://github.com/microsoft/playwright/pull/26730/commits/01a3f7180c7c014d3f09790c5136f78b9c154033